src/sdk/pynni/nni/compression/torch/speedup/infer_shape.py

```diff
@@ -626,64 +843,69 @@ def convert_to_coarse_mask(mask):
         """
         assert 'weight' in mask
         assert isinstance(mask['weight'], torch.Tensor)
+        assert dim in [0, 1]
+
         weight_mask = mask['weight']
-        shape = weight_mask.size()
-        ones = torch.ones(shape[1:]).to(weight_mask.device)
-        zeros = torch.zeros(shape[1:]).to(weight_mask.device)
-        index = []
-        for i in range(shape[0]):
-            if torch.all(torch.eq(weight_mask[i], ones)):
-                index.append(i)
-            elif torch.all(torch.eq(weight_mask[i], zeros)):
-                continue
-            else:
-                index = None
-                break
+
+        sum_idx = (1, 2, 3) if dim == 0 else (0, 2, 3)
+        index = torch.nonzero(weight_mask.abs().sum(sum_idx) != 0, as_tuple=True)[0]
+        if len(index) == weight_mask.shape[dim]: # full mask
+            index = None
+
         if index is None:
             return None, None, None
         else:
             index = torch.LongTensor(index).to(weight_mask.device)
             weight_cmask = CoarseMask(num_dim=4)
-            weight_cmask.add_index_mask(dim=0, index=index)
+            weight_cmask.add_index_mask(dim=dim, index=index)
             bias_cmask = None
-            if 'bias' in mask and mask['bias'] is not None:
+            if dim == 0 and 'bias' in mask and mask['bias'] is not None:
                 bias_index = torch.nonzero(mask['bias'], as_tuple=True)[0]
                 assert torch.all(torch.eq(index, bias_index)), \
                     "bias mask should be consistent with weight mask"
                 bias_cmask = CoarseMask(num_dim=1)
                 bias_cmask.add_index_mask(dim=0, index=bias_index)
             return index, weight_cmask, bias_cmask
-    index, weight_cmask, bias_cmask = convert_to_coarse_mask(mask)
+
+    index, weight_cmask, bias_cmask = convert_to_coarse_mask(mask, dim=conv_prune_dim)
+
     if index is None:
         # TODO: fine grained mask speedup
         return None, None
     # deal with coarse grain mask
+    # mask conflict should be solved by fix_mask_conflict before speedup
     if 'weight' in module_masks.param_masks:
-        module_masks.param_masks['weight'].merge(weight_cmask)
-        module_masks.param_masks['bias'].merge(bias_cmask)
```

Comment:
I think we should merge the mask here, but I'm not very clearly, we can discuss later.

For example, 
conv1 and conv2 are two adjacent layers. conv2 prunes the input channel(4,5,6) and conv1 prunes the output channel(1,2,3).

Reply:
as we discussed, we should not allow both input and output being pruned, it is checked in detect_prune_dim.